main.js

```diff
@@ -0,0 +1,107 @@
+var VideoView = Backbone.View.extend({
+  className: 'current_video',
+
+  template: Handlebars.compile($('#current-video-template').html()),
+  
+  render: function () {
+    this.$el.html(this.template(this.model.toJSON()));
+
+    return this;
+  }
+});
+
+var VideosView = Backbone.View.extend({
+  className: 'video_list',
+
+  template: Handlebars.compile($('#video-list-template').html()),
+  
+  render: function () {
+    this.$el.html(this.template(this.model.toJSON()));
+
+    return this;
+  }
+});
+
+var VideoModel = Backbone.Model.extend({
+  defaults: function () {
+    return {
+      title: '',
+      description: '',
+      videoId: '',
+      image_url: '',
+    }
+  },
+
+  urlRoot: 'https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q=cats&type=video&videoEmbeddable=true&key=AIzaSyBMMdn63cR3Sby4uyuYiEWH-xltbXvmGZc',
+
+  parse: function (response) {
+    console.log(response);
+    return {
+      title: response.items[0].snippet.title,
+      description: response.items[0].snippet.description,
+      videoId: response.items[0].id.videoID,
+      image_url: response.items[0].snippet.thumbnails.default.url
+    }
+  }
+});
+
+var VideoCollection = Backbone.Collection.extend({
+  model: VideoModel,
+
+  initialize: function () {
+    this.on('add', function(model) {
+      model.fetch();
+    });
+  }
+});
+
+var AppModel = Backbone.Model.extend({
+  defaults: function () {
+    return {
+      videos: new VideoCollection()
+    }
+  }
+});
+
+var AppView = Backbone.View.extend({
+  el: $('.app-container'),
+
+  events: {
+    'click .search': 'handleSearchClick'
+  },
+
+  initialize: function () {
+    this.$searchInput = this.$('#search-query');
+    this.$video = this.$('video');
+    this.$videos = this.$('.videos');
+
+    this.listenTo(this.model.get('video'), 'change', this.renderVideo);
+  },
+
+  handleSearchClick: function() {
+    var query = this.$searchInput.val();
+
+    this.model.get('videos').add({
```

Comment:
This where things we wrong. Calling `add` on a collection adds a new model to it. So here you're merely pushing an object with `query` into it. Instead you want to add the new videos (so push a new video model).

However, you don't have any videos yet. At this point, I'd encourage you to review the rules of data flow. 

When a user clicks the button, we want to fetch the videos. This means we need to call `fetch` on the videos collection - you're doing it on the model.

As the videos come back, we can listen for `add` events from the AppView (or we can pass `reset: true` to `fetch` and listen for a reset event in the AppView). We can listen for those `add` (or `reset`) event(s) from the AppView and call some render function(s) to actually render the data.